OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -160,7 +160,7 @@ class MyCoursesListFragment : OfflineSupportBaseFragment(), RefreshListener {
     }
 
     private fun initInAppPurchaseSetup() {
-        if (environment.remoteFeaturePrefs.isValuePropEnabled()) {
+        if (environment.remoteFeaturePrefs.isValuePropEnabled() && isAdded) {
```

Comment:
better to rearrange the conditions like `if (isAdded && environment.remoteFeaturePrefs.isValuePropEnabled())`.